src/manipulation/buildFragment.js

```diff
@@ -35,15 +37,17 @@ function buildFragment( elems, context, scripts, selection, ignored ) {
 
 				// Deserialize a standard representation
 				tag = ( rtagName.exec( elem ) || [ "", "" ] )[ 1 ].toLowerCase();
-				wrap = wrapMap[ tag ] || wrapMap._default;
-				tmp.innerHTML = wrap[ 1 ] + jQuery.htmlPrefilter( elem ) + wrap[ 2 ];
+				wrap = wrapMap[ tag ] || arr;
 
-				// Descend through wrappers to the right content
-				j = wrap[ 0 ];
-				while ( j-- ) {
-					tmp = tmp.lastChild;
+				// Create wrappers & descend into them.
+				j = wrap.length;
+				while ( --j > -1 ) {
+					tmp.appendChild( document.createElement( wrap[ j ] ) );
+					tmp = tmp.firstChild;
```

Comment:
I think this needs to use `context` rather than `document`, and will probably be smaller as one statement rather than two.
```suggestion
					tmp = tmp.appendChild( context.createElement( wrap[ j ] ) );
```